BUILD.md

```diff
@@ -99,22 +99,41 @@ it on the CMake command line for building this repository, as described below.
 
 ### Building Dependent Repositories with Known-Good Revisions
 
-There is a Python utility script, `scripts/update_deps.py`, that you can use
-to gather and build the dependent repositories mentioned above. This program
-also uses information stored in the `scripts/known-good.json` file to checkout
-dependent repository revisions that are known to be compatible with the
-revision of this repository that you currently have checked out.
+There is a Python utility script, `scripts/update_deps.py`, that you can use to
+gather and build the dependent repositories mentioned above. This script uses
+information stored in the `scripts/known_good.json` file to check out dependent
+repository revisions that are known to be compatible with the revision of this
+repository that you currently have checked out. As such, this script is useful
+as a quick-start tool for common use cases and default configurations.
 
-Here is a usage example for this repository:
+For all platforms, start with:
 
     git clone https://github.com/LunarG/VulkanSamples.git
     cd VulkanSamples
     mkdir build
     cd build
+
+For 64-bit Linux and MacOS, continue with:
+
     ../scripts/update_deps.py
     cmake -C helper.cmake ..
     cmake --build .
 
+For 64-bit Windows, continue with:
+
+    ..\scripts\update_deps.py --arch x64
+    cmake -A x64 -C helper.cmake ..
+    cmake --build .
+
+For 32-bit Windows, continue with:
+
+    ..\scripts\update_deps.py --arch Win32
```

Comment:
Oh, and I approved with the review... I'd like to see this changed but I'm fine if you decide not to change it.

Reply:
I copied the VVL commit for consistency.  It's the same in VulkanTools, too.

Reply:
> I'd just look for parallelism. If we use x64 for 64-bit, then we should use x86 for 32-bit. If we use Win64 for 64-bit, then we should use Win32 for 32-bit.

Parallelism?  Do you mean consistency?

The problem is that the [-A option ](https://cmake.org/cmake/help/v3.15/generator/Visual%20Studio%2016%202019.html)for CMake takes either `Win32` or `x64`.

update_deps lets you say `32`, `x86`, or `Win32` if you want a 32-bit build.  The last one was intentionally put in place to "match" the CMake -A option that selects a 32-bit build.

This all lets you say:
```
..\scripts\update_deps.py --arch Win32
cmake -A Win32 -C helper.cmake ..
cmake --build .
```

where you get to say `Win32` in each place for consistency.

Would you rather force or suggest someone say:
```
..\scripts\update_deps.py --arch x86
cmake -A Win32 -C helper.cmake ..
cmake --build .
```

???

It's unfortunate that CMake chose "Win32" for a -A option because "Win32" is often used to refer to the Windows API, which exists and is called "Win32" on 32-bit and 64-bit Windows.  In this context, it is an API and not an architecture.  But there are other contexts where "Win32" specifies 32-bit executables and libraries and "Win64" specifies the 64-bit equivalents.

But yeah, it would have been better if CMake had decided to use x86 for the -A option or at least added it later as an alias.  Perhaps you could make a feature request over at CMake.  Then we could specify x86 or x64 in both the update_deps and CMake invocations.

I think that CMake was trying to match what Visual Studio _was_ doing.  In VS2013, if you create a new Windows Application, you get the "Win32" flavor by default and you can add the x64 one if you want.  If you do the same in VS2017, it creates the solution/project with x86 and x64 flavors.  I think we're simply dealing with Windows legacy stuff here.